src/metrics/test/metrics_test.cpp

```diff
@@ -502,5 +508,122 @@ TEST_CASE("Registry counts the number of workers") {
     REQUIRE(metrics.getWorkerCount("actor2", "op1") == 2);
 }
 
+TEST_CASE("Events stream to gRPC") {
+    using EventVec = std::vector<poplar::EventMetrics>;
+
+    auto compareEventsAndClear = [](const EventVec& events_in) {
+        internals::v2::MockStreamInterface interface;
+        REQUIRE(events_in.size() == interface.events.size());
+        for (int i = 0; i < events_in.size(); i++) {
+            REQUIRE(google::protobuf::util::MessageDifferencer::Equals(events_in[i],
+                                                                       interface.events[i]));
+            // REQUIRE(2 == 1);
+        };
+        interface.events.clear();
+    };
+
+    SECTION("Empty stream") {
+        RegistryClockSourceStub::reset();
+        auto stream =
+            internals::v2::EventStream<RegistryClockSourceStub, internals::v2::MockStreamInterface>{
+                1, "TestName", 1, "/test/prefix"};
+
+        EventVec expected;
+        compareEventsAndClear(expected);
+    }
+
+    SECTION("Two events") {
+        RegistryClockSourceStub::reset();
+        auto stream =
+            internals::v2::EventStream<RegistryClockSourceStub, internals::v2::MockStreamInterface>{
+                1, "EventName", 9, "/test/prefix"};
+        EventVec expected;
+
+
+        // First Event
+        RegistryClockSourceStub::advance(std::chrono::microseconds(5));
+        {
+            OperationEventT<RegistryClockSourceStub> event(
+                2,
+                3,
+                7,
+                0,
```

Comment:
I wish C++ had lightweight named params here! Can you add inline comments about what these fields are

```
event(
  2, // duration
  3, // size
```